apps/sensenet/src/components/field-controls/webhook-filter.tsx

```diff
@@ -0,0 +1,564 @@
+      const newSelected = typesSelected.filter((type) => item !== type)
+      setTypesSelected(newSelected)
+
+      setValue({ ...value, ContentTypes: value.ContentTypes?.filter((type) => item !== type.Name) })
+      props.fieldOnChange?.(
+        props.settings.Name,
+        JSON.stringify({ ...value, ContentTypes: value.ContentTypes?.filter((type) => item !== type.Name) }),
+      )
+    }
+  }
+
+  const handleCheckboxChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    contentType: string,
+    webhookEvent: WebhookEventType,
+  ) => {
+    const copyOfContentTypes = value.ContentTypes
+    const actualRow = copyOfContentTypes?.find((type) => type.Name === contentType)
+
+    if (event.target.checked) {
+      !actualRow?.Events.includes(webhookEvent) && actualRow?.Events.push(webhookEvent)
+    } else {
+      const actualEventIndex = actualRow?.Events.indexOf(webhookEvent)
+      if (actualEventIndex !== undefined && actualEventIndex > -1) {
+        actualRow?.Events.splice(actualEventIndex, 1)
+      }
+
+      const allEventIndex = actualRow?.Events.indexOf('All')
+      if (allEventIndex !== undefined && allEventIndex > -1) {
+        const newEvents = webhookEvents.filter((item) => item.name !== webhookEvent)
+        Object.assign(actualRow, { Events: newEvents })
+      }
+    }
+
+    setValue({ ...value, ContentTypes: copyOfContentTypes })
+    props.fieldOnChange?.(props.settings.Name, JSON.stringify(value))
+  }
+
+  const handleCheckboxAllChange = (event: React.ChangeEvent<HTMLInputElement>, contentType: string) => {
+    const copyOfContentTypes = value.ContentTypes
+    const actualRow = copyOfContentTypes?.find((type) => type.Name === contentType)
+
+    if (event.target.checked) {
+      Object.assign(actualRow, { Events: ['All'] })
+    } else {
+      Object.assign(actualRow, { Events: [] })
+    }
+
+    setValue({ ...value, ContentTypes: copyOfContentTypes })
+    props.fieldOnChange?.(props.settings.Name, JSON.stringify(value))
+  }
+
+  const handleDialogClose = () => {
+    setIsPickerOpen(false)
+  }
+
+  const renderTypeListItem = (item: string, select: (item: string) => void) => (
+    <ListItem key={item} value={item} onClick={() => select(item)}>
+      <ListItemIcon style={{ margin: 0 }}>{renderIconDefault(typeicons.contenttype)}</ListItemIcon>
+      <ListItemText primary={item} />
+    </ListItem>
+  )
+
+  switch (props.actionName) {
+    case 'edit':
+    case 'new':
+      return (
+        <>
+          <div className={classes.containerSelector}>
+            <TextField
+              className={classes.containerSelectorInput}
+              autoFocus={props.autoFocus}
+              autoComplete="off"
+              name={props.settings.Name}
+              id={props.settings.Name}
+              label={props.settings.DisplayName}
+              placeholder={props.settings.DisplayName}
+              value={value.Path}
+              fullWidth={true}
+            />
```

Comment:
should be readonly

Reply:
disabled? It is not the right way I think :D 
![image](https://user-images.githubusercontent.com/22347393/111339981-7f028080-8678-11eb-9035-3af4c9d1bd81.png)